### PR TITLE
Small fixes and typing

### PR DESCRIPTION
**Pull request by andrewtarzia**
Requested Reviewers: @lukasturcani

This PR will be slow going, but I will occasionally do folders in tests and update their doc strings to Google and add TypeHints.

Also fixed issues with turbomole writer (allows `pathlib.Path`) and the default behaviour of `stk.host_guest.Guest`

### CODE REVIEW

File: tests/molecular/molecules/building_block/test_get_num_functional_groups.py [hunk 1]
```diff
@@ -1,16 +1,18 @@
...
-    ----------
-    case_data : :class:`.CaseData`
-        A test case. Holds the building block to test and the
-        correct number of functional groups.
+    Parameters:
+        case_data:
+            A test case. Holds the building block to test and the correct
+            functional groups.
+
+    Returns:
```
Comment (lukasturcani) [inline tests/molecular/molecules/building_block/test_get_num_functional_groups.py:13]: you can rmeove the returns section

File: tests/molecular/molecules/building_block/test_get_placer_ids.py [hunk 2]
```diff
@@ -23,21 +24,22 @@ def test_get_placer_ids(case_data):
...
 
-    placer_ids : :class:`tuple` of :class:`int`
-        The correct placer ids.
+        placer_ids:
+            The correct placer ids.
 
-    Returns
-    -------
-    None : :class:`NoneType`
+    Returns:
```
Comment (lukasturcani) [inline tests/molecular/molecules/building_block/test_get_placer_ids.py:41]: you can remove the returns sectoin

File: tests/molecular/molecules/building_block/test_repr.py [hunk 3]
```diff
@@ -1,16 +1,18 @@
...
-        A test case. Holds the building block to test and the correct
-        functional groups.
+    Parameters:
+        case_data:
+            A test case. Holds the building block to test.
 
-    Returns
-    -------
-    None : :class:`NoneType`
+    Returns:
```
Comment (lukasturcani) [inline tests/molecular/molecules/building_block/test_repr.py:14]: you can remove the returns section

File: tests/molecular/molecules/building_block/test_repr.py [hunk 4]
```diff
@@ -20,22 +22,21 @@ def test_repr(case_data):
...
 
-    functional_groups : :class:`tuple` of :class:`.FunctionalGroup`
-        The correct functional groups.
+        known_repr:
+            The correct representation.
 
-    Returns
-    -------
-    None : :class:`NoneType`
+    Returns:
```
Comment (lukasturcani) [inline tests/molecular/molecules/building_block/test_repr.py:36]: you can remove the returns sectoin

File: tests/molecular/molecules/building_block/test_with_functional_groups.py [hunk 5]
```diff
@@ -4,24 +4,27 @@
...
-        Takes a single parameter, `building_block` and returns the
-        `functional_groups` parameter to use for this test.
+        get_functional_groups:
+            Takes a single parameter, `building_block` and returns the
+            `functional_groups` parameter to use for this test.
 
-    Returns
-    -------
-    None : :class:`NoneType`
+    Returns:
```
Comment (lukasturcani) [inline tests/molecular/molecules/building_block/test_with_functional_groups.py:26]: remove returns section

File: tests/molecular/molecules/building_block/test_with_functional_groups.py [hunk 6]
```diff
@@ -35,21 +38,22 @@ def test_with_functional_groups(building_block, get_functional_groups):
...
 
-    functional_groups : :class:`tuple` of :class:`.FunctionalGroup`
-        The functional groups the new building block should hold.
+        functional_groups:
+            The functional groups the new building block should hold.
 
-    Returns
-    -------
-    None : :class:`NoneType`
+    Returns:
```
Comment (lukasturcani) [inline tests/molecular/molecules/building_block/test_with_functional_groups.py:55]: remove

File: tests/molecular/writers/mdl_mol/case_data.py [hunk 7]
```diff
@@ -1,34 +1,40 @@
+import stk
+
+
 class CaseData:
```
Comment (lukasturcani) [inline tests/molecular/writers/mdl_mol/case_data.py:4]: use data class

File: tests/molecular/writers/mdl_mol/test_to_string.py [hunk 8]
```diff
@@ -1,15 +1,18 @@
...
-    case_data : :class:`.CaseData`
-        A test case.
+    Parameters:
+        case_data:
+            A test case.
 
-    Returns
-    -------
-    None : :class:`NoneType`
+    Returns:
```
Comment (lukasturcani) [inline tests/molecular/writers/mdl_mol/test_to_string.py:14]: remove

File: tests/molecular/writers/mdl_mol/test_to_string.py [hunk 9]
```diff
@@ -20,24 +23,26 @@ def test_to_string(case_data):
...
 
-    string : :class:`str`
-        The expected output string.
+        string:
+            The expected output string.
 
-    Returns
-    -------
-    None : :class:`NoneType`
+    Returns:
```
Comment (lukasturcani) [inline tests/molecular/writers/mdl_mol/test_to_string.py:44]: remove

File: tests/molecular/writers/mdl_mol/test_write.py [hunk 10]
```diff
@@ -1,18 +1,22 @@
...
 
-    tmp_path : :class:`pathlib2.Path`
-        Path to temporary directory.
+        tmp_path:
+            Path to temporary directory.
 
-    Returns
-    -------
-    None : :class:`NoneType`
+    Returns:
```
Comment (lukasturcani) [inline tests/molecular/writers/mdl_mol/test_write.py:18]: remove

File: tests/molecular/writers/mdl_mol/test_write.py [hunk 11]
```diff
@@ -24,27 +28,30 @@ def test_write(case_data, tmp_path):
...
 
-    file_path : :class:`str`
-        Path to temporary file.
+        file_path:
+            Path to temporary file.
 
-    Returns
-    -------
-    None : :class:`NoneType`
+    Returns:
```
Comment (lukasturcani) [inline tests/molecular/writers/mdl_mol/test_write.py:53]: remove

File: tests/molecular/writers/pdb/case_data.py [hunk 12]
```diff
@@ -1,41 +1,48 @@
+import stk
+
+
 class CaseData:
```
Comment (lukasturcani) [inline tests/molecular/writers/pdb/case_data.py:4]: use dataclass

File: tests/molecular/writers/pdb/test_to_string.py [hunk 13]
```diff
@@ -1,15 +1,18 @@
...
-    case_data : :class:`.CaseData`
-        A test case.
+    Parameters:
+        case_data:
+            A test case.
 
-    Returns
-    -------
-    None : :class:`NoneType`
+    Returns:
```
Comment (lukasturcani) [inline tests/molecular/writers/pdb/test_to_string.py:14]: remove

File: tests/molecular/writers/pdb/test_to_string.py [hunk 14]
```diff
@@ -21,27 +24,30 @@ def test_to_string(case_data):
...
 
-    periodic_info : :class:`.PeriodicInfo`
-        Periodic information.
+        periodic_info:
+            Periodic information.
 
-    Returns
-    -------
-    None : :class:`NoneType`
+    Returns:
```
Comment (lukasturcani) [inline tests/molecular/writers/pdb/test_to_string.py:49]: remove

File: tests/molecular/writers/pdb/test_write.py [hunk 15]
```diff
@@ -26,35 +30,33 @@ def test_write(case_data, tmp_path):
...
 
-    file_path : :class:`str`
-        Path to temporary file.
+        periodic_info:
+            Periodic information.
 
-    Returns
-    -------
-    None : :class:`NoneType`
+    Returns:
```
Comment (lukasturcani) [inline tests/molecular/writers/pdb/test_write.py:58]: remove

File: tests/molecular/writers/turbomole/case_data.py [hunk 16]
```diff
@@ -1,41 +1,48 @@
+import stk
+
+
 class CaseData:
```
Comment (lukasturcani) [inline tests/molecular/writers/turbomole/case_data.py:4]: dataclass

File: tests/molecular/writers/turbomole/test_to_string.py [hunk 17]
```diff
@@ -1,15 +1,18 @@
...
-    case_data : :class:`.CaseData`
-        A test case.
+    Parameters:
+        case_data:
+            A test case.
 
-    Returns
-    -------
-    None : :class:`NoneType`
+    Returns:
```
Comment (lukasturcani) [inline tests/molecular/writers/turbomole/test_to_string.py:14]: remove

File: tests/molecular/writers/turbomole/test_to_string.py [hunk 18]
```diff
@@ -21,27 +24,30 @@ def test_to_string(case_data):
...
 
-    periodic_info : :class:`.PeriodicInfo`
-        Periodic information.
+        periodic_info:
+            Periodic information.
 
-    Returns
-    -------
-    None : :class:`NoneType`
+    Returns:
```
Comment (lukasturcani) [inline tests/molecular/writers/turbomole/test_to_string.py:49]: remove

File: tests/molecular/writers/turbomole/test_write.py [hunk 19]
```diff
@@ -1,21 +1,24 @@
...
 
-    tmp_path : :class:`pathlib2.Path`
-        Path to temporary directory.
+        tmp_path:
+            Path to temporary directory.
 
-    Returns
-    -------
-    None : :class:`NoneType`
+    Returns:
```
Comment (lukasturcani) [inline tests/molecular/writers/turbomole/test_write.py:18]: remove

File: tests/molecular/writers/turbomole/test_write.py [hunk 20]
```diff
@@ -26,35 +29,33 @@ def test_write(case_data, tmp_path):
...
 
-    file_path : :class:`str`
-        Path to temporary file.
+        periodic_info:
+            Periodic information.
 
-    Returns
-    -------
-    None : :class:`NoneType`
+    Returns:
```
Comment (lukasturcani) [inline tests/molecular/writers/turbomole/test_write.py:57]: remove

File: tests/molecular/writers/xyz/case_data.py [hunk 21]
```diff
@@ -1,34 +1,40 @@
+import stk
+
+
 class CaseData:
```
Comment (lukasturcani) [inline tests/molecular/writers/xyz/case_data.py:4]: dataclass

File: tests/molecular/writers/xyz/test_to_string.py [hunk 22]
```diff
@@ -1,15 +1,18 @@
...
-    case_data : :class:`.CaseData`
-        A test case.
+    Parameters:
+        case_data:
+            A test case.
 
-    Returns
-    -------
-    None : :class:`NoneType`
+    Returns:
```
Comment (lukasturcani) [inline tests/molecular/writers/xyz/test_to_string.py:14]: remove

File: tests/molecular/writers/xyz/test_to_string.py [hunk 23]
```diff
@@ -20,24 +23,26 @@ def test_to_string(case_data):
...
 
-    string : :class:`str`
-        The expected output string.
+        string:
+            The expected output string.
 
-    Returns
-    -------
-    None : :class:`NoneType`
+    Returns:
```
Comment (lukasturcani) [inline tests/molecular/writers/xyz/test_to_string.py:44]: remove

File: tests/molecular/writers/xyz/test_write.py [hunk 24]
```diff
@@ -1,18 +1,22 @@
...
 
-    tmp_path : :class:`pathlib2.Path`
-        Path to temporary directory.
+        tmp_path:
+            Path to temporary directory.
 
-    Returns
-    -------
-    None : :class:`NoneType`
+    Returns:
```
Comment (lukasturcani) [inline tests/molecular/writers/xyz/test_write.py:18]: remove

File: tests/molecular/writers/xyz/test_write.py [hunk 25]
```diff
@@ -24,27 +28,30 @@ def test_write(case_data, tmp_path):
...
 
-    file_path : :class:`str`
-        Path to temporary file.
+        file_path:
+            Path to temporary file.
 
-    Returns
-    -------
-    None : :class:`NoneType`
+    Returns:
```
Comment (lukasturcani) [inline tests/molecular/writers/xyz/test_write.py:53]: remove

File: src/stk/_internal/topology_graphs/host_guest/complex.py [hunk 26]
```diff
@@ -449,6 +449,13 @@ class Complex(TopologyGraph):
                 ),
             )
 
+    Warnings:
+
+        The host of a :class:`.Complex` will always be placed at the
+        origin, not at the centroid of the input building block. Therefore,
+        to place a guest at the centroid of the host use
+        `displacement=(0, 0, 0)`, which is **now** the default behaviour.
```
[the remark this answers is not in the document]
Comment (andrewtarzia) [reply src/stk/_internal/topology_graphs/host_guest/complex.py:457]: The host is moved to the origin (the first sentence states this, line 510 does too: `building_block_vertices = {host: [HostVertex(0, (0.0, 0.0, 0.0))]}`). Changes made.

File: tests/molecular/molecules/building_block/case_data.py [hunk 27]
```diff
@@ -1,54 +1,56 @@
+import stk
+
+
 class CaseData:
     """
     A test case.
+    Attributes:
```
[the remark this answers is not in the document]
Comment (andrewtarzia) [reply tests/molecular/molecules/building_block/case_data.py:7]: Done

File: tests/molecular/molecules/building_block/case_data.py [hunk 28]
```diff
@@ -1,54 +1,56 @@
+import stk
+
+
 class CaseData:
```
[the remark this answers is not in the document]
Comment (andrewtarzia) [reply tests/molecular/molecules/building_block/case_data.py:4]: Done.

File: tests/molecular/molecules/building_block/conftest.py [hunk 29]
```diff
@@ -48,18 +50,17 @@ def building_block(case_data):
         ),
     )
 )
-def get_functional_groups(request):
+def get_functional_groups(
```
[the remark this answers is not in the document]
Comment (andrewtarzia) [reply tests/molecular/molecules/building_block/conftest.py:53]: I use `abc.Iterable` for yield statements, is that ok?

File: tests/molecular/molecules/building_block/fixtures/init_from_file.py [hunk 30]
```diff
@@ -17,78 +15,76 @@ class InitFromFileData:
...
-        building_block,
-        init_functional_groups,
-        init_placer_ids,
-        case_data_functional_groups,
-        case_data_core_atom_ids,
-        case_data_placer_ids,
-        known_repr,
-    ):
+        building_block: stk.BuildingBlock,
+        init_functional_groups: Iterable,
```
[the remark this answers is not in the document]
Comment (andrewtarzia) [reply tests/molecular/molecules/building_block/fixtures/init_from_file.py:51]: Can you explain this one?

File: tests/molecular/molecules/building_block/fixtures/init_from_file.py [hunk 31]
```diff
@@ -17,78 +15,76 @@ class InitFromFileData:
...
+        init_placer_ids: tuple | None,
+        case_data_functional_groups: tuple,
```
[the remark this answers is not in the document]
Comment (andrewtarzia) [reply tests/molecular/molecules/building_block/fixtures/init_from_file.py:53]: Can you explain this one?

**Review by lukasturcani — APPROVED** [review]

**Pull request merged**